lib/s3.js

```diff
@@ -0,0 +1,72 @@
+const S3 = require('aws-sdk/clients/s3')
+const s3 = new S3({ apiVersion: '2006-03-01' })
+
+/**
+ * select - select data from S3 using S3-select
+ *
+ * @param {object} params object to hold bucket name, key, input/output serialization etc.
+ * @param {object} options
+ * @return {Array<Object>}      Array of objects as selected data from s3
+ */
+async function select (params, options = {}) {
+  let buffer = Buffer.alloc(0)
+
+  return new Promise((resolve, reject) => {
+    s3.selectObjectContent(params, (err, data) => {
+      if (err) {
+        return reject(err)
+      }
+
+      if (!data) {
+        return reject(new Error('No data recieved'))
+      }
+
+      // data.Payload is a Readable Stream
+      const eventStream = data.Payload
+
+      // Read events as they are available
+      eventStream.on('data', (event) => {
+        // event.Records.Payload is a buffer containing a single record, partial records, or multiple records so concatenate buffers
```

Comment:
you may want to consider using async/await to process the stream e.g.
```
async function main(inputFilePath) {
  const readStream = fs.createReadStream(inputFilePath,
    { encoding: 'utf8', highWaterMark: 1024 });

  for await (const chunk of readStream) {
    console.log('>>> '+chunk);
  }
  console.log('### DONE ###');
}
```

Reply:
When I change to this pattern, I get the follow warning:
```
(node:22518) ExperimentalWarning: Readable[Symbol.asyncIterator] is an experimental feature. This feature could change at any time
```
The code works, but wondering if we should still use event handlers until `Symbol.asyncIterator` is past experimental?

Reply:
this is on node 10?

Reply:
Ack....VsCode was defaulting to 9.8.  Good on 10

Reply:
I have updated using `for await` and adjusted unit tests accordingly.